include/zephyr/drivers/clock_control/stm32_clock_control.h

```diff
@@ -148,6 +148,19 @@
 #define STM32_SYSCLK_SRC_IC2	1
 #endif
 
+#if defined(CONFIG_SOC_SERIES_STM32N6X)
```

Comment:
Even if equivalent today in this case, I prefer we avoid the use of `CONFIG_SOC_SERIES_FOO` and seek for use of compatibles (`DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(rcc), st_stm32n6_rcc, okay)`).

This should ultimately help to factorize code as much as possible.